pkg/repository/merge_mirror.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repository
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiup/pkg/repository/model"
+	"github.com/pingcap/tiup/pkg/repository/v0manifest"
+	"github.com/pingcap/tiup/pkg/repository/v1manifest"
+	"github.com/pingcap/tiup/pkg/version"
+)
+
+type diffItem struct {
+	name          string
+	componentItem v1manifest.ComponentItem
+	versionItem   v1manifest.VersionItem
+	version       string
+	os            string
+	arch          string
+	desc          string
+}
+
+// returns component that exists in source but not in base
```

Comment:
"source" and "base" are bad names, they have similar meaning and might be confusing. how about "base" and "input" or "addition"